core/src/snapshot_utils.rs

```diff
@@ -185,26 +239,25 @@ pub fn untar_snapshot_in<P: AsRef<Path>, Q: AsRef<Path>>(
     Ok(())
 }
 
-fn hardlink_snapshot_directory<P: AsRef<Path>, Q: AsRef<Path>>(
-    snapshot_dir: P,
-    snapshot_hardlink_dir: Q,
-    slot: u64,
-) -> Result<()> {
-    // Create a new directory in snapshot_hardlink_dir
-    let new_slot_hardlink_dir = snapshot_hardlink_dir.as_ref().join(slot.to_string());
-    let _ = fs::remove_dir_all(&new_slot_hardlink_dir);
-    fs::create_dir_all(&new_slot_hardlink_dir)?;
-
-    // Hardlink the contents of the directory
-    let snapshot_file = snapshot_dir
-        .as_ref()
-        .join(slot.to_string())
-        .join(slot.to_string());
-    fs::hard_link(
-        &snapshot_file,
-        &new_slot_hardlink_dir.join(slot.to_string()),
-    )?;
-    Ok(())
+impl SlotSnapshotPaths {
```

Comment:
Move this impl to below struct definition?

Reply:
will do on rebase.